gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/KafkaJobStatusMonitor.java

```diff
@@ -140,7 +143,27 @@ private void addJobStatusToStateStore(org.apache.gobblin.configuration.State job
     String jobGroup = jobStatus.getProp(TimingEvent.FlowEventConstants.JOB_GROUP_FIELD);
     String storeName = jobStatusStoreName(flowGroup, flowName);
     String tableName = jobStatusTableName(flowExecutionId, jobGroup, jobName);
-    this.stateStore.put(storeName, tableName, jobStatus);
+
+    jobStatus = mergedProperties(storeName, tableName, jobStatus, stateStore);
+
+    stateStore.put(storeName, tableName, jobStatus);
+  }
+
+  private static org.apache.gobblin.configuration.State mergedProperties(
```

Comment:
Can you add JavaDoc explaining the behavior of this method?

Reply:
Any reason why this method is static? I do not see it being called from outside the class.

Reply:
being called from a test class